### PR TITLE
Replace hc::__lane_id() with __lane_id() for hip-clang

### PR DESCRIPTION
**Pull request by yxsamliu**

### CODE REVIEW

**Comment by ROCm-Apps-Test** [comment]
Can one of the admins verify this patch?

**Comment by ROCm-Apps-Test** [comment]
Can one of the admins verify this patch?

File: tensorflow/core/util/gpu_device_functions.h [hunk 1]
```diff
@@ -125,7 +125,7 @@ __device__ inline unsigned GpuLaneId() {
   asm("mov.u32 %0, %%laneid;" : "=r"(lane_id));
 #elif TENSORFLOW_USE_ROCM
   // ROCM TODO add ROCM implementation
-  lane_id = hc::__lane_id();
```
Comment (whchung) [inline tensorflow/core/util/gpu_device_functions.h]: @yxsamliu are you able to verify if this change works with the incumbent hip/hcc toolchain?

Comment (yxsamliu) [reply tensorflow/core/util/gpu_device_functions.h]: It builds with hip/hcc.

Which tests should I run to verify?

**Comment by whchung** [comment]
jenkins: retest this please

**Comment by whchung** [comment]
jenkins: retest this please

**Pull request merged**